test/integration/router_container_test.rb

```diff
@@ -13,4 +13,17 @@
     response = @app.get('/back/home', lint: true)
     response.body.must_equal 'back'
   end
+
+  it 'matches against host' do
+    @router_container = Hanami::Router.new(scheme: 'https', host: 'hanami.test', port: 443) do
+      mount Front::App, at: '/front', host: 'www.hanami.test'
```

Comment:
@nicolaracco What happens if a developer only uses the subdomain? Eg. `host: 'www'`.